x-pack/plugins/siem/public/components/page/hosts/kpi_hosts/index.tsx

```diff
@@ -0,0 +1,166 @@
+        icon: 'cross',
+      },
+    ],
+    enableAreaChart: true,
+    enableBarChart: true,
+    grow: 4,
+    description: i18n.AUTHENTICATION,
+  },
+  {
+    fields: [
+      {
+        key: 'uniqueSourceIps',
+        name: i18n.UNIQUE_SOURCE_IPS_ABBREVIATION,
+        description: i18n.UNIQUE_SOURCE_IPS,
+        value: null,
+        color: euiColorVis2,
+        icon: 'visMapCoordinate',
+      },
+      {
+        key: 'uniqueDestinationIps',
+        description: i18n.UNIQUE_DESTINATION_IPS,
+        value: null,
+        color: euiColorVis3,
+        icon: 'visMapCoordinate',
+      },
+    ],
+    enableAreaChart: true,
+    enableBarChart: true,
+    grow: 4,
+    description: i18n.UNIQUE_IPS,
+  },
+];
+
+export const KpiHostsComponent = pure<KpiHostsProps>(({ data, loading }) => {
+  return loading ? (
+    <EuiFlexGroup justifyContent="center" alignItems="center" style={{ minHeight: 247 }}>
+      <EuiFlexItem grow={false}>
+        <EuiLoadingSpinner size="xl" />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  ) : (
+    <EuiFlexGroup>
+      {fieldTitleMapping.map(stat => {
+        let statItemProps: StatItemsProps = {
+          ...stat,
+          key: `kpi-hosts-summary-${stat.description}`,
+        };
+
+        if (stat.fields != null)
+          statItemProps = {
+            ...statItemProps,
+            fields: addValueToFields(stat.fields, data),
+          };
+
+        if (stat.enableAreaChart)
+          statItemProps = {
+            ...statItemProps,
+            areaChart: addValueToAreaChart(stat.fields, data),
+          };
+
+        if (stat.enableBarChart != null)
+          statItemProps = {
+            ...statItemProps,
+            barChart: addValueToBarChart(stat.fields, data),
+          };
+
+        return <StatItemsComponent {...statItemProps} />;
+      })}
+    </EuiFlexGroup>
+  );
+});
+
+const addValueToFields = (fields: StatItem[], data: KpiHostsData): StatItem[] =>
+  fields.map(field => ({ ...field, value: get(field.key, data) }));
+
+const addValueToAreaChart = (fields: StatItem[], data: KpiHostsData): AreaChartData[] =>
+  fields
+    .filter(field => get(`${field.key}Histogram`, data) != null)
+    .map(field => ({
+      ...field,
+      value: get(`${field.key}Histogram`, data),
+      key: `${field.key}Histogram`,
+    }));
+
+const addValueToBarChart = (fields: StatItem[], data: KpiHostsData): BarChartData[] => {
+  if (fields.length === 0) return [];
+  return fields.reduce((acc: BarChartData[], field: StatItem, idx: number) => {
+    const key = get('key', field);
+    const x: number | null = getOr(null, key, data);
+    const y: string = get(`${idx}.name`, fields) || getOr('', `${idx}.description`, fields);
+    const dataSet: BarChartData[] = [];
+    if (y != null)
+      dataSet.push({
+        ...field,
+        value: [
+          {
+            x,
+            y,
+          },
+        ],
+      });
+    return acc.concat(dataSet);
+  }, []);
```

Comment:
Optional: I think you can you eliminate the `push` and extra array with:

```ts
    if (y != null) {
      return acc.concat([
        {
          ...field,
          value: [
            {
              x,
              y,
            },
          ],
        },
      ]);
    } else {
      return acc.concat([]);
    }
```

Also, your type for `y` above looks like it should never be `null`?

```ts
const y: string = get(`${idx}.name`, fields) || getOr('', `${idx}.description`, fields);
```
So wondering if you need the `if (y != null)` call or not? If so, I would change the type to be `const y: string | null | undefined = get(`)